src/oci-umount.c

```diff
@@ -365,10 +365,15 @@ static int unmount(char *umount_path, bool submounts_only, const struct mount_in
 
 	if (!submounts_only) {
 		ret = umount2(umount_path, MNT_DETACH);
-		if (!ret)
-			pr_pinfo("Unmounted: [%s]\n", umount_path);
-		else
-			pr_perror("Failed to unmount: [%s]\n", umount_path);
+		if (ret) {
+			if (errno == EINVAL) {
+				pr_pdebug("%s: Not a mount point [%s]\n", id, umount_path);
```

Comment:
How does EINVAL always imply that it is not a valid mount point.  Also before we call unmount() we already check if umount_path() is a mount point or not. So this change does not look right to me.

Reply:
Man page.
man umount2 

       EINVAL target is not a mount point.

       EINVAL umount2() was called with MNT_EXPIRE and either MNT_DETACH or MNT_FORCE.

       EINVAL (since Linux 2.6.34)
              umount2() was called with an invalid flag value in flags.

Since the other two would only happen at coding time, most likely the programmer would catch it.